Mathlib/Analysis/NormedSpace/Basic.lean

```diff
@@ -514,3 +514,46 @@ def NormedAlgebra.restrictScalars : NormedAlgebra 𝕜 E :=
 end NormedAlgebra
 
 end RestrictScalars
+
+section Core
+
+/-- A structure encapsulating minimal axioms needed to defined a normed vector space, as found
+in textbooks. This is meant to be used to easily define `NormedAddCommGroup E` and `NormedSpace E`
+instances from scratch on a type with no preexisting distance. -/
+structure NormedSpace.Core (𝕜 : Type*) (E : Type*) [NormedField 𝕜] [AddCommGroup E] [Module 𝕜 E]
+    [Norm E] : Prop where
+  norm_nonneg (x : E) : 0 ≤ ‖x‖
+  norm_eq_zero_iff (x : E) : ‖x‖ = 0 ↔ x = 0
```

Comment:
Can we drop this condition from `.Core`? Then we can have a `SeminormedAddCommGroup.ofCore` and a `NormedAddCommGroup.ofCore` where the latter requires additionally this hypothesis.

Reply:
I just did this. It does feel a bit silly to have to have module stuff in something called `SeminormedAddCommGroup.Core` but I don't have a better naming idea. Also, I don't see how to avoid doing the proofs twice, is there a clean way to reuse the seminormed structure in the normed one?